MarvelApiClient/src/main/java/com/karumi/MarvelApiClient/MarvelApiClient.java

```diff
@@ -28,97 +29,151 @@
+      } catch (IOException e) {
+      }
     }
 
-    /**
-     * Fluent API for creating {@link MarvelApiClient} instances.
-     */
-    @SuppressWarnings("UnusedDeclaration")
-    public static class Builder {
+    throw new MarvelApiException(execute.code(), marvelCode, marvelDescription, null);
+  }
+
+  /**
+   * Fluent API for creating {@link MarvelApiClient} instances.
+   */
+  @SuppressWarnings("UnusedDeclaration") public static class Builder {
 
-        private static final String MARVEL_URL = "http://gateway.marvel.com/v1/public/";
-        private final String privateKey;
-        private final String publicKey;
-        private boolean debug;
-        private Retrofit retrofit;
-        private String baseUrl = MARVEL_URL;
+    private static final String MARVEL_URL = "http://gateway.marvel.com/v1/public/";
+    private final String privateKey;
+    private final String publicKey;
+    private boolean debug;
+    private Retrofit retrofit;
+    private String baseUrl = MARVEL_URL;
+    private TimeProvider timeProvider;
+
+    public Builder(String publicKey, String privateKey) {
+      if (publicKey == null) {
+        throw new IllegalArgumentException("publicKey must not be null.");
+      }
+
+      if (privateKey == null) {
+        throw new IllegalArgumentException("privateKey must not be null.");
+      }
+
+      this.publicKey = publicKey;
+      this.privateKey = privateKey;
+    }
 
+    public Builder debug() {
+      this.debug = true;
+      return this;
+    }
 
-        public Builder(String publicKey, String privateKey) {
-            if (publicKey == null) {
-                throw new IllegalArgumentException("publicKey must not be null.");
-            }
+    public Builder baseUrl(String url) {
+      this.baseUrl = url;
+      return this;
+    }
 
-            if (privateKey == null) {
-                throw new IllegalArgumentException("privateKey must not be null.");
-            }
+    public Builder retrofit(Retrofit retrofit) {
+      if (retrofit == null) {
+        throw new IllegalArgumentException("retrofit must not be null.");
+      }
+      this.retrofit = retrofit;
+      return this;
+    }
 
-            this.publicKey = publicKey;
-            this.privateKey = privateKey;
-        }
+    public Builder timeProvider(TimeProvider timeProvider) {
+      if (timeProvider == null) {
+        throw new IllegalArgumentException("timeProvider must not be null");
+      }
+      this.timeProvider = timeProvider;
+      return this;
+    }
 
-        public Builder debug(boolean debug) {
-            this.debug = debug;
-            return this;
-        }
+    public MarvelApiClient build() {
+      if (retrofit == null) {
+        retrofit = createDefaultRetrofit(baseUrl, debug);
+      }
 
-        public Builder baseUrl(String url) {
-            this.baseUrl = url;
-            return this;
-        }
+      if (timeProvider == null) {
+        timeProvider = new SystemTimeProvider();
```

Comment:
Ok, interface give the posibility to create a Fake Test double easilly if we need simulate time increases, for this code is not necessary right now, I can change it.